bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -689,6 +701,38 @@ TEST_F(BcStTest, dstValidateRealSourceListReported) {
   ASSERT_EQ(BCStateTran::FetchingState::NotFetching, stateTransfer_->getFetchingState());
 }
 
+// Validate a recurring source selection, during ongoing state transfer;
+TEST_F(BcStTest, validatePeriodicSourceReplacement) {
+  ASSERT_NO_FATAL_FAILURE(SendCheckpointSummaries());
+  mockedSrc_->ReplyAskForCheckpointSummariesMsg();
+  int batch_count = 0;
+  while (true) {
+    ASSERT_NO_FATAL_FAILURE(
+        AssertFetchBlocksMsgSent(testParams_.expectedFirstRequiredBlockNum, testParams_.expectedLastRequiredBlockNum));
+    // once the source is selected, adding sleep for more than source replacement time duration
+    if (batch_count < 2) {
+      this_thread::sleep_for(milliseconds(config_.sourceReplicaReplacementTimeoutMs));
+    }
+    mockedSrc_->ReplyFetchBlocksMsg();
+    if (testParams_.expectedLastRequiredBlockNum <= config_.maxNumberOfChunksInBatch) break;
+    testParams_.expectedLastRequiredBlockNum -= config_.maxNumberOfChunksInBatch;
+    // There might be pending jobs for putBlock, we need to wait some time and then finalize them by calling
+    this_thread::sleep_for(chrono::milliseconds(20));
+    onTimerImp();
+    batch_count++;
+  }
+  const auto& sources_ = GetSourceSelector().getActualSources();
+  ASSERT_EQ(sources_.size(), 3);
```

Comment:
I think we need to expect some computed number, in case we increase number of blocks fetched or change periodic replacment config param. Test should be adjustable to configuration.
In that case I would expect the number of sources to be:
time to send batches / periodic replacment time + 1